src/utils/paths.rs

```diff
@@ -154,7 +154,9 @@ fn get_native_path(filetype: FileType, identifier: &ObjectId) -> Option<String>
                 ObjectType::Elf => get_gdb_path(identifier)?,
                 ObjectType::Unknown => return None,
             };
-            base_path.truncate(base_path.len() - base_path.rsplitn(2, '.').next()?.len() - 1);
+            base_path.truncate(
+                (base_path.len() - base_path.rsplitn(2, '.').next()?.len()).checked_sub(1)?,
```

Comment:
pretty sure what you want here is:

```rust
if let Some(cutoff) = base_path.rfind('.') {
    base_path.truncate(cutoff);
}
```

Reply:
Also, probably this should protect against dots in the path, since file names may contain dots. Paths are hard :D

Reply:
For copy/paste convenience:

```rust
            if let Some(cutoff) = base_path.rfind('.') {
                if cutoff > base_path.rfind('/').unwrap_or(0) {
                    base_path.truncate(cutoff);
                }
            }
```

(note that this changes test assertions, rightfully so.